src/hotspot/share/prims/jvmtiThreadState.inline.hpp

```diff
@@ -82,7 +82,9 @@ inline JvmtiThreadState* JvmtiThreadState::state_for_while_locked(JavaThread *th
   JvmtiThreadState *state = thread == nullptr ? nullptr : thread->jvmti_thread_state();
 
   if (state == nullptr && thread != nullptr &&
-      (thread->is_exiting() || thread->is_attaching_via_jni())) {
+      (thread->is_exiting() ||
+       (thread->threadObj() == nullptr && thread->is_attaching_via_jni())
+      )) {
     // Don't add a JvmtiThreadState to a thread that is exiting or is attaching.
     // When a thread is attaching, it may not have a Java level thread object
     // created yet.
```

Comment:
The comment needs adjusting now - suggestion:
```
// Don't add a JvmtiThreadState to a thread that is exiting, or is attaching
// and does not yet have a Java level thread object allocated.
```

Reply:
Thanks for the suggested change. I agree and I've applied it.